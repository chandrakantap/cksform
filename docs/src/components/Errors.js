import React from 'react';

export default function ({ errors = {}, fieldName }) {
    const fieldErrors = errors[fieldName];
    if (!Array.isArray(fieldErrors) || fieldErrors.length === 0) {
        return null;
    }
    return (
      <ul className="error">
        {fieldErrors.map((error, index) => (<li key={index}>{error}</li>))}
      </ul>
    )
}