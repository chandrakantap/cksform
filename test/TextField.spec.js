import React from 'react';
import { mount } from 'enzyme';
import { createSerializer } from 'enzyme-to-json';
import useCKSForm from '../src/index';
import Errors from '../docs/src/components/Errors';

expect.addSnapshotSerializer(createSerializer({ mode: 'deep', noKey: true, ignoreDefaultProps: true }));

const TextFieldForm = ({ initialValue = '', onSubmit, ...props }) => {
    const { formData, onChangeField, errors, handleSubmit, validateField } = useCKSForm({ title: initialValue });
    return (
      <form noValidate autoComplete="off">
        <input
          type="text"
          name="title"
          required
          minLength={6}
          value={formData.title}
          onChange={onChangeField}
          onBlur={validateField}
          data-display-name="Title"
          {...props}
        />
        <Errors errors={errors} fieldName="title" />
        <input id="submitBtn" type="button" value="Submit" onClick={handleSubmit(onSubmit)} />
      </form>
    );
}
describe('Input field of type text', () => {
    test('renders properly', () => {
        const warapper = mount(<TextFieldForm />);
        expect(warapper).toMatchSnapshot();
    });

    test('Validates when no value provided', () => {
        const warapper = mount(<TextFieldForm />);
        warapper.find("input[type='button']").first().simulate('click');
        expect(warapper).toMatchSnapshot();
    });
})
