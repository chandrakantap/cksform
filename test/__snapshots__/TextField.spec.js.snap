// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input field of type text Validates when no value provided 1`] = `
<form
  autoComplete="off"
  noValidate={true}
>
  <input
    data-display-name="Title"
    minLength={6}
    name="title"
    onBlur={[Function]}
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
  <ul
    className="error"
  >
    <li>
      Title is required.
    </li>
  </ul>
  <input
    id="submitBtn"
    onClick={[Function]}
    type="button"
    value="Submit"
  />
</form>
`;

exports[`Input field of type text renders properly 1`] = `
<form
  autoComplete="off"
  noValidate={true}
>
  <input
    data-display-name="Title"
    minLength={6}
    name="title"
    onBlur={[Function]}
    onChange={[Function]}
    required={true}
    type="text"
    value=""
  />
  <input
    id="submitBtn"
    onClick={[Function]}
    type="button"
    value="Submit"
  />
</form>
`;
